src/Tribe/Admin/Pages.php

```diff
@@ -0,0 +1,265 @@
+
+		return $this->current_page;
+	}
+
+	/**
+	 * Connect an existing page to wc-admin.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $options {
+	 *   Array describing the page.
+	 *
+	 *   @type string       id           Id to reference the page.
+	 *   @type string|array title        Page title. Used in menus and breadcrumbs.
+	 *   @type string|null  parent       Parent ID. Null for new top level page.
+	 *   @type string       path         Path for this page. E.g. admin.php?page=wc-settings&tab=checkout
+	 *   @type string       capability   Capability needed to access the page.
+	 *   @type string       icon         Icon. Dashicons helper class, base64-encoded SVG, or 'none'.
+	 *   @type int          position     Menu item position.
+	 * }
+	 */
+	public function connect_page( $options ) {
+		if ( ! is_array( $options['title'] ) ) {
+			$options['title'] = array( $options['title'] );
+		}
+
+		/**
+		 * Filter the options when connecting or registering a page.
+		 *
+		 * @param array $options {
+		 *   Array describing the page.
+		 *
+		 *   @type string       id           Id to reference the page.
+		 *   @type string|array title        Page title. Used in menus and breadcrumbs.
+		 *   @type string|null  parent       Parent ID. Null for new top level page.
+		 *   @type string       screen_id    The screen ID that represents the connected page. (Not required for registering).
+		 *   @type string       path         Path for this page. E.g. admin.php?page=wc-settings&tab=checkout
+		 *   @type string       capability   Capability needed to access the page.
+		 *   @type string       icon         Icon. Dashicons helper class, base64-encoded SVG, or 'none'.
+		 *   @type int          position     Menu item position.
+		 *   @type boolean      js_page      If this is a JS-powered page.
+		 * }
+		 */
+		$options = apply_filters( 'tec_admin_pages_connect_page_options', $options );
+
+		$this->pages[ $options['id'] ] = $options;
+	}
+
+	/**
+	 * Get the capability.
+	 *
+	 * @since TBD
+	 *
+	 * @return string|null
+	 */
+	public static function get_capability( $capability = 'manage_options' ) {
+		/**
+		 * Filters the default capability for Tribe admin pages.
+		 *
+		 * @todo: We'll need to deprecate this one in favor of the one below.
+		 */
+		$capability = apply_filters( 'tribe_common_event_page_capability', $capability );
+
+		/**
+		 * Filters the default capability for TEC admin pages.
+		 *
+		 * @since TBD
+		 */
+		$capability = apply_filters( 'tec_admin_pages_capability', $capability );
+
+		return $capability;
+	}
+
+	/**
+	 * Define if is a `tec` admin page (registered).
+	 *
+	 * @return boolean
+	 */
+	public function is_tec_page() {
+		// @todo @juanfra: Check if it's part of the array of `tec_admin_pages`.
+		return true;
+	}
+
+	/**
+	 * Get pages with tabs.
+	 *
+	 * @param array $pages The list of pages with tabs.
+	 * @return array $pages The list of pages with tabs, filtered.
```

Comment:
```suggestion
	 * @since TBD
	 *
	 * @param array $pages The list of pages with tabs.
	 * @return array $pages The list of pages with tabs, filtered.
```